torch/_functorch/aot_autograd.py

```diff
@@ -2285,19 +2292,42 @@ def compute_overlapping_inputs(fwd_inputs, aliased_input_indices):
     return actual_aliased_indices
 
 
-def _check_if_mutation_can_be_in_graph(keep_input_mutations: bool, is_train: bool, mutates_data, mutates_metadata, requires_grad):
+def _check_if_mutation_can_be_in_graph(
+    keep_input_mutations: bool,
+    is_train: bool,
+    mutates_data,
+    mutates_metadata,
+    mutations_hidden_from_autograd,
+    requires_grad
+):
     if keep_input_mutations:
+        if mutations_hidden_from_autograd:
```

Comment:
oh - this should probably be `mutates_data and (not requires_grad or mutations_hidden_from_autograd)`?

`mutates_data` --> there was some data mutation on the input
`not requires_grad or mutations_hidden_from_autograd` -> if the input didn't require_grad, keep the input mutation in the graph. BUT if the input did required grad, and every mutation that it saw was hidden from autograd, then also keep the input mutation in the graph.